client/a8c-for-agencies/components/onboarding-tour-modal/index.tsx

```diff
@@ -0,0 +1,24 @@
+import { Modal } from '@wordpress/components';
+import { Children, isValidElement, ReactNode } from 'react';
+import OnboardingTourModalSection from './section';
+
+interface OnboardingTourModalProps {
+	onClose: () => void;
+	children?: React.ReactNode;
+}
+
+function OnboardingTourModal( { onClose, children }: OnboardingTourModalProps ) {
+	const sections = Children.toArray( children ).filter(
+		( child: ReactNode ) => isValidElement( child ) && child.type === OnboardingTourModalSection
+	);
+
+	return (
+		<Modal className="onboarding-tour-modal" onRequestClose={ onClose } __experimentalHideHeader>
+			<div className="onboarding-tour-modal__main">{ sections }</div>
+		</Modal>
```

Comment:
This is just the initial implementation. We will develop the DOM structure in a separate PR.